.github/workflows/deploy-lib.yml

```diff
@@ -4,6 +4,20 @@ on:
   push:
     branches:
       - main
+    paths:
```

Comment:
These should also include paths to the workflow, so that if the workflow changes they will be ran